apps/studio/data/api-keys/temp-api-keys-utils.ts

```diff
@@ -47,9 +48,7 @@ const checkOrRefreshTemporaryApiKey = async (
 // This function should never be marked as async, it should always return a promise.
 export function getOrRefreshTemporaryApiKey(projectRef: ProjectRef): Promise<TemporaryApiKey> {
   const existingKey = projectApiKeys.get(projectRef)
-  if (!existingKey) {
-    const data = checkOrRefreshTemporaryApiKey(projectRef, undefined)
-    projectApiKeys.set(projectRef, data)
-  }
-  return projectApiKeys.get(projectRef)!
+  const data = checkOrRefreshTemporaryApiKey(projectRef, existingKey)
+  projectApiKeys.set(projectRef, data)
+  return data
```

Comment:
@ivasilov it would only check if the Key is expired if the key didn't exist haha. We need to check it each time we get the key